[#38028] Refresh button for Revit Add-in

wielinde opened the pull request:
https://community.openproject.org/work_packages/38028

- The buttons only shows up when OP is loaded in the Revit Add-in.
- It reloads the whole route to achieve two things:
  - Reload queries when there is a WP list/cards visible
  - Reload a WP when a specific WP is open.

wielinde committed:
[#38028] Refresh button for Revit Add-in

https://community.openproject.org/work_packages/38028

oliverguenther commented on frontend/src/app/features/bim/ifc_models/pages/viewer/ifc-viewer-page.component.ts:
```diff
@@ -58,6 +59,10 @@ export class IFCViewerPageComponent extends PartitionedQuerySpacePageComponent {
     },
+    {
+      component: RefreshButtonComponent,
+      show: () => window.navigator.userAgent.search('Revit') > -1,
```
You should probably use the `ViewerBridgeService` which already has this logic defined to check whether the button is used.

oliverguenther commented on frontend/src/app/features/bim/ifc_models/toolbar/import-export-bcf/refresh-button.component.ts:
```diff
@@ -0,0 +1,63 @@
+  `,
+  selector: 'refresh-button',
+})
+export class RefreshButtonComponent extends UntilDestroyedMixin implements OnInit, OnDestroy {
```
UntilDestroyedMixin can be removed

wielinde replied:
I also remove `implements OnInit, OnDestroy` as we actually don't implement it.

oliverguenther commented on frontend/src/app/features/bim/ifc_models/toolbar/import-export-bcf/refresh-button.component.ts:
```diff
@@ -0,0 +1,63 @@
+
+  constructor(readonly I18n:I18nService,
+    readonly state:StateService) {
+    super();
```
super() call can be removed without `UntilDestroyedMixin`

oliverguenther commented on frontend/src/app/features/bim/ifc_models/toolbar/import-export-bcf/refresh-button.component.ts:
```diff
@@ -0,0 +1,63 @@
+  }
+
+  refresh() {
+    this.state.go('.', {}, { reload: true });
```
Didn't you change this to reload the work package instead? Is this the desired behavior?

wielinde replied:
Yes, I found out that I also need to be able to refresh the WP list. This way it works for both scenarios and I don't need to check any contexts like "am I in the list?" or "am I in the WP?" or "both?". Probably that could be better but it does the job.

oliverguenther commented on frontend/src/app/features/bim/ifc_models/toolbar/import-export-bcf/refresh-button.component.ts:
```diff
@@ -0,0 +1,63 @@
+    super();
+  }
+
+  ngOnInit() {
```
ngOnInit can be removed

wielinde replied:
done

oliverguenther commented on frontend/src/app/features/bim/ifc_models/toolbar/import-export-bcf/refresh-button.component.ts:
```diff
@@ -0,0 +1,63 @@
+      <op-icon icon-classes="button--icon icon-workflow"></op-icon>
+    </a>
+  `,
+  selector: 'refresh-button',
```
Selector should be `op-refresh-button` to match our styleguide

wielinde replied:
You are right. Thank you.

wielinde committed:
Improvements to BCF modules refresh button (post review)

Also
- Reuse ViewerBridgeService for checking client context (Revit or not)
- Align selector naming to our convention
- Remove all OnInit, OnDestroy, and UntilDestroyedMixin

wielinde committed:
Satisfy eslint
